api/files/serializers.py

```diff
@@ -416,6 +425,14 @@ class FileDetailSerializer(FileSerializer):
     """
     id = IDField(source='_id', required=True)
 
+    def to_representation(self, value):
```

Comment:
can we access the request and match the ID to the one used? we shouldn't return the guid if the request was to `v2/files/_id`, for instance. Similarly, we should default to the internal id and not the guid any time we're doing a list.

Reply:
I don't entirely understand, this should return the non-guid id by default because `.load` will return None, making the default `to_representation` behavior happens, but `v2/files/_id` should throw a 404 in the view, without making it to the serializer. 

Also I don't understand about getting the id directly from the request, we have easier access to that data from the view context without reading the url string.

Reply:
Nothing about this PR changes whether getting `/v2/files/_id` would 404 or not, and https://api.staging2.osf.io/v2/files/620ea23613649b0010dbdfd6/ works properly. Accessing that url should have an `id` of 620ea23613649b0010dbdfd6 while accessing the same file via its guid https://api.staging2.osf.io/v2/files/tzgay/ should have an `id` of tzgay.

Reply:
Wow, yeah, did not read closely enough. Thought that this was just always trying to load a guid for the file and using that if it existed; missed that we're just attempting to load the provided id to see if it happens to be a guid and then overriding the default non-guid id.